File: 14_longest_common_prefix/main.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]

        for i in range(1, len(strs)):
            while strs[i].find(prefix) != 0:
                if prefix:
                    prefix = prefix[:-1]
                else:
                    prefix = ""

                if not prefix:
                    return ""
        return prefix

File: 14_longest_common_prefix/main.go
package main

import "fmt"

func find(pattern string, text string) int {
	window_size := len(pattern)
	text_len := len(text)

	for i := 0; i < (text_len - window_size + 1); i++ {
		if pattern == text[i:window_size+i] {
			return i
		}
	}
	return -1
}

func longestCommonPrefix(strs []string) string {
	base := strs[0]

	for i := 1; i < len(strs); i++ {
		for find(base, strs[i]) != 0 {
			if base != "" {
				base = base[:len(base)-1]
			} else {
				base = ""
			}
		}
	}
	return base
}

func main() {
	strs := []string{"c", "acc", "ccc"}
	result := longestCommonPrefix(strs)
	fmt.Println(result)

	// result := find("fl", "flower")
	// fmt.Println(result)

}
